### PR TITLE
mtd_spi_nor: make thread safe

### PR DESCRIPTION
**Pull request by pwillem**
### Contribution description
This commit moves `spi_release`/`spi_acquire` 1 level higher so the spi lock is only release when a complete transaction happened. Without this fix it could be that based on thread priority an other thread could interrupt an ongoing read/write operation.

### CODE REVIEW

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 1]
```diff
@@ -87,14 +87,11 @@ static void mtd_spi_cmd_addr_read(const mtd_spi_nor_t *dev, uint8_t opcode,
     uint8_t *addr_buf = &addr.u8[4 - dev->addr_width];
     if (ENABLE_TRACE) {
         TRACE("mtd_spi_cmd_addr_read: addr:");
-        for (unsigned int i = 0; i < dev->addr_width; ++i)
-        {
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 2]
```diff
@@ -103,10 +100,7 @@ static void mtd_spi_cmd_addr_read(const mtd_spi_nor_t *dev, uint8_t opcode,
 
         /* Read data */
         spi_transfer_bytes(dev->spi, dev->cs, false, NULL, dest, count);
-    } while(0);
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 3]
```diff
@@ -129,14 +123,11 @@ static void mtd_spi_cmd_addr_write(const mtd_spi_nor_t *dev, uint8_t opcode,
     uint8_t *addr_buf = &addr.u8[4 - dev->addr_width];
     if (ENABLE_TRACE) {
         TRACE("mtd_spi_cmd_addr_write: addr:");
-        for (unsigned int i = 0; i < dev->addr_width; ++i)
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 4]
```diff
@@ -147,10 +138,7 @@ static void mtd_spi_cmd_addr_write(const mtd_spi_nor_t *dev, uint8_t opcode,
         if (cont) {
             spi_transfer_bytes(dev->spi, dev->cs, false, (void *)src, NULL, count);
         }
-    } while(0);
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 5]
```diff
@@ -165,14 +153,9 @@ static void mtd_spi_cmd_addr_write(const mtd_spi_nor_t *dev, uint8_t opcode,
 static void mtd_spi_cmd_read(const mtd_spi_nor_t *dev, uint8_t opcode, void* dest, uint32_t count)
 {
     TRACE("mtd_spi_cmd_read: %p, %02x, %p, %" PRIu32 "\n",
-        (void *)dev, (unsigned int)opcode, dest, count);
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 6]
```diff
@@ -187,14 +170,9 @@ static void mtd_spi_cmd_read(const mtd_spi_nor_t *dev, uint8_t opcode, void* des
 static void __attribute__((unused)) mtd_spi_cmd_write(const mtd_spi_nor_t *dev, uint8_t opcode, const void* src, uint32_t count)
 {
     TRACE("mtd_spi_cmd_write: %p, %02x, %p, %" PRIu32 "\n",
-        (void *)dev, (unsigned int)opcode, src, count);
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 7]
```diff
@@ -207,14 +185,9 @@ static void __attribute__((unused)) mtd_spi_cmd_write(const mtd_spi_nor_t *dev,
 static void mtd_spi_cmd(const mtd_spi_nor_t *dev, uint8_t opcode)
 {
     TRACE("mtd_spi_cmd: %p, %02x\n",
-        (void *)dev, (unsigned int)opcode);
```
Comment (vincent-d) [inline drivers/mtd_spi_nor/mtd_spi_nor.c]: unrelated

**Comment by pwillem** [comment]
Moved all unrelated changes to a new commit
